Addon/Context/NoFrameworkRenderingContext.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using EPiServer.Core;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using RenderingLayoutProcessor.Extension;
+
+namespace RenderingLayoutProcessor.Context
+{
+    public class NoFrameworkRenderingContext : IRenderingContentAreaContext
+    {
+        public IRenderingContentAreaContext ParentContext { get; set; }
+
+        protected bool _enableContainerWrapper;
+
+        public NoFrameworkRenderingContext()
+        {
+        }
+
+        public void ContainerOpen(IHtmlHelper htmlHelper)
+        {
+            if (_enableContainerWrapper)
+            {
+                ProcessContainer(htmlHelper);
+            }
+
+            var rowTag = new TagBuilder("div");
+
+            foreach (var rowClass in GetRowClasses())
+            {
+                rowTag.AddCssClass(rowClass);
+            }
+            var blockMetaData = htmlHelper.BlockMetaData();
+            rowTag.MergeAttribute("data-layout", blockMetaData.LayoutContentLink.ID.ToString());
+            rowTag.MergeAttribute("data-layout-index", blockMetaData.LayoutIndex.ToString());
+            rowTag.RenderOpenTo(htmlHelper);
+        }
+
+        private void ProcessContainer(IHtmlHelper htmlHelper)
+        {
+            var containerTag = new TagBuilder("div");
+
+            foreach (var containerClass in GetContainerClasses())
+            {
+                containerTag.AddCssClass(containerClass);
+            }
+
+            containerTag.RenderOpenTo(htmlHelper);
+        }
+
+        public void ItemOpen(IHtmlHelper htmlHelper)
+        {
+        }
+
+        protected virtual IEnumerable<string> GetContainerClasses()
+        {
+            return Enumerable.Empty<string>();
+        }
+
+        protected virtual IEnumerable<string> GetRowClasses()
+        {
+            return Enumerable.Empty<string>();
+        }
+
+        protected virtual IEnumerable<string> GetColumnClasses()
+        {
+            return Enumerable.Empty<string>();
+        }
+
+        public RenderingProcessorAction RenderItem(IHtmlHelper htmlHelper, ContentAreaItem current, Action renderItem)
+        {
+            renderItem();
+            return RenderingProcessorAction.Close;
```

Comment:
This will force so there can only be one block per layout block. Is that the intent?

Reply:
In my testing that's not the behavior I observe. 
On the test page for NoFramework there is multiple layouts and multiple blocks as well. 

![image](https://user-images.githubusercontent.com/1185533/170995809-0459f61c-2924-46c3-b53b-932752e50773.png)